tasks/main.yml

```diff
@@ -160,3 +160,56 @@
   when: galaxy_manage_cleanup
   tags:
     - galaxy_manage_cleanup
+
+- name: Create welcome.html directory for basedomain
+  ansible.builtin.file:
+    state: directory
+    mode: '0755'
+    owner: "{{ __galaxy_privsep_user_name }}"
+    group: "{{ __galaxy_privsep_user_group }}"
+    path: "{{ galaxy_themes_static_path }}/static/welcome.html"
+  when: galaxy_manage_subdomain_static
+
+- name: Template welcome.html for basedomain
+  ansible.builtin.template:
+    src: welcome.html.j2
```

Comment:
I think all of the `when: galaxy_manage_subdomain_static` could be moved into a separate file to match the rest of `main.yml` that does conditional includes.